File: src-tauri/src/domain/response/saldo.rs
use crate::model::saldo::{
    SaldoMonthBalanceResponse, SaldoMonthTotalBalanceResponse, SaldoResponse,
    SaldoResponseDeleteAt, SaldoYearBalanceResponse, SaldoYearTotalBalanceResponse,
};

use super::PaginationMeta;
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize)]
pub struct ApiResponseSaldo {
    pub status: String,
    pub message: String,
    pub data: SaldoResponse,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponsesSaldo {
    pub status: String,
    pub message: String,
    pub data: Vec<SaldoResponse>,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponseSaldoDelete {
    pub status: String,
    pub message: String,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponseSaldoAll {
    pub status: String,
    pub message: String,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponseMonthTotalSaldo {
    pub status: String,
    pub message: String,
    pub data: Vec<SaldoMonthTotalBalanceResponse>,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponseYearTotalSaldo {
    pub status: String,
    pub message: String,
    pub data: Vec<SaldoYearTotalBalanceResponse>,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponseMonthSaldoBalances {
    pub status: String,
    pub message: String,
    pub data: Vec<SaldoMonthBalanceResponse>,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponseYearSaldoBalances {
    pub status: String,
    pub message: String,
    pub data: Vec<SaldoYearBalanceResponse>,
}

#[derive(Serialize, Deserialize, Debug)]
pub struct ApiResponsePaginationSaldo {
    pub status: String,
    pub message: String,
    pub data: Vec<SaldoResponse>,
    pub pagination: PaginationMeta,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponsePaginationSaldoDeleteAt {
    pub status: String,
    pub message: String,
    pub data: Vec<SaldoResponseDeleteAt>,
    pub pagination: PaginationMeta,
}

File: src/types/domain/response/saldo/saldo.ts
import {
  Saldo,
  SaldoMonthBalance,
  SaldoMonthTotalBalance,
  SaldoTrashed,
  SaldoYearBalance,
  SaldoYearTotalBalance,
} from "@/types/model";
import { PaginationMeta } from "../pagination";

export interface ApiResponseSaldo {
  status: string;
  message: string;
  data: Saldo;
}

export interface ApiResponsesSaldo {
  status: string;
  message: string;
  data: Saldo[];
}

export interface ApiResponseSaldoDelete {
  status: string;
  message: string;
}

export interface ApiResponseSaldoAll {
  status: string;
  message: string;
}

export interface ApiResponseMonthTotalSaldo {
  status: string;
  message: string;
  data: SaldoMonthTotalBalance[];
}

export interface ApiResponseYearTotalSaldo {
  status: string;
  message: string;
  data: SaldoYearTotalBalance[];
}

export interface ApiResponseMonthSaldoBalances {
  status: string;
  message: string;
  data: SaldoMonthBalance[];
}

export interface ApiResponseYearSaldoBalances {
  status: string;
  message: string;
  data: SaldoYearBalance[];
}

export interface ApiResponsePaginationSaldo {
  status: string;
  message: string;
  data: Saldo[];
  pagination: PaginationMeta;
}

export interface ApiResponsePaginationSaldoDeleteAt {
  status: string;
  message: string;
  data: SaldoTrashed[];
  pagination: PaginationMeta;
}
